client/my-sites/themes/controller.js

```diff
@@ -1,97 +1,153 @@
-	setSection = require( 'state/ui/actions' ).setSection,
-	getCurrentUser = require( 'state/current-user/selectors' ).getCurrentUser,
-	buildTitle = require( 'lib/screen-title/utils' ),
-	getAnalyticsData = require( './helpers' ).getAnalyticsData;
-
-var controller = {
-
-	themes: function( context ) {
-		const { tier, site_id } = context.params;
-		const user = getCurrentUser( context.store.getState() );
-		const title = buildTitle(
-			i18n.translate( 'Themes', { textOnly: true } ),
-			{ siteID: site_id } );
-		const Head = user
-			? require( 'layout/head' )
-			: require( 'my-sites/themes/head' );
-
-		let ThemesComponent;
-
-		if ( user ) {
-			ThemesComponent = site_id ? SingleSiteComponent : MultiSiteComponent;
-		} else {
-			ThemesComponent = LoggedOutComponent;
-		}
-
-		const { basePath, analyticsPageTitle } = getAnalyticsData(
-			context.path,
-			tier,
-			site_id
-		);
+import SingleSiteComponent from 'my-sites/themes/single-site';
+import MultiSiteComponent from 'my-sites/themes/multi-site';
+import LoggedOutComponent from 'my-sites/themes/logged-out';
+import { ThemeSheet as ThemeSheetComponent } from 'my-sites/themes/sheet';
+import analytics from 'analytics';
+import i18n from 'lib/mixins/i18n';
+import trackScrollPage from 'lib/track-scroll-page';
+import buildTitle from 'lib/screen-title/utils';
+import { getAnalyticsData } from './helpers';
+import { getCurrentUser } from 'state/current-user/selectors';
+import { setSection } from 'state/ui/actions';
+import ClientSideEffects from './client-side-effects';
+
+function getProps( context ) {
+	const { tier, site_id: siteId } = context.params;
+
+	const title = buildTitle(
+		i18n.translate( 'Themes', { textOnly: true } ),
+		{ siteID: siteId } );
 
+	const { basePath, analyticsPageTitle } = getAnalyticsData(
+		context.path,
+		tier,
+		siteId
+	);
+
+	const runClientAnalytics = function() {
 		analytics.pageView.record( basePath, analyticsPageTitle );
-		ReactDom.render(
-			React.createElement( ReduxProvider, { store: context.store },
-				React.createElement( Head, { title, tier: tier || 'all' },
-					React.createElement( ThemesComponent, {
-						key: site_id,
-						siteId: site_id,
-						tier: tier,
-						search: context.query.s,
-						trackScrollPage: trackScrollPage.bind(
-							null,
-							basePath,
-							analyticsPageTitle,
-							'Themes'
-						)
-					} )
-				)
-			),
-			document.getElementById( 'primary' )
+	};
+
+	const boundTrackScrollPage = function() {
+		trackScrollPage(
+			basePath,
+			analyticsPageTitle,
+			'Themes'
 		);
-	},
+	};
 
-	details: function( context ) {
-		const user = getCurrentUser( context.store.getState() );
+	return {
+		title,
+		tier,
+		search: context.query.s,
+		trackScrollPage: boundTrackScrollPage,
+		runClientAnalytics
+	};
+}
 
-		const Head = user
-			? require( 'layout/head' )
-			: require( 'my-sites/themes/head' );
+function createElement( ThemesComponent, Head, store, props ) {
```

Comment:
Seeing `createElement` in the controllers threw me off a little. Maybe something different? Ideas: `createElementTree`, `createWrappedElement`, `makeElements`, etc.